packages/webpack/src/config/base.js

```diff
@@ -443,7 +441,7 @@ export default class WebpackBaseConfig {
     const { extend } = this.buildContext.buildOptions
     if (typeof extend === 'function') {
       const extendedConfig = extend.call(
-        this.builder, config, { loaders: this.loaders, ...this.nuxtEnv }
+        this.builder, config, { babelLoader: this.getBabelLoader(), loaders: this.loaders, ...this.nuxtEnv }
```

Comment:
What's the point of creating a new babel loader object and passing it to extend here?

Reply:
OK. I saw the intention in the comments but still, this is making an unnecessary object when not needed. Can't we extract babel loader config from `config`?

Reply:
@pi0 The thing is that the babel loader config depends on `this.getBabelOptions` which are dynamic depending on if it's `modern` mode or not (with class inheritance), and they also depend on config build options.

What should we do then ? I don't know how we can extract it.

All I know is that the final result need to make `babelLoader` config available in Nuxt modules.

Reply:
The module can use this in `extend` to extract specific babel config:

```js
const babelLoader = config.module.rules.find(r => r.test.toString() === '/\\.jsx?$/i').use[0]
```

Seems little hacky but prevents calling `getBabelLoader()` multiple times...

----

Actually, my concern is just that duplicate call... What do you think about these possible options:

1. Compute `babelLoader` inside the base constructor and pass it to the context like `{ babelLoader: this.babelLoader }`
2. Pass `{ webpackConfig: this, loaders: ... }`
3. The `find` method proposed above
4. Keep it as is. Ignoring little `getBabelLoader()` overhead

/cc @clarkdo

Reply:
Even if hacky I like the 3rd one.
As the statement is `use: perfLoader.js().concat(babelLoader)` I guess it should be `use[use.length - 1]` ?

Reply:
And honestly has the babelLoader is need for a `tsx` context, it sounds not so hacky to grab babelLoader with `jsx` regex :)

Reply:
Reverted babelLoader changes

Reply:
https://github.com/nuxt/typescript/blob/master/packages/typescript/lib/module.js#L33